### ISSUE TITLE
Async publishing always delivers messages

### ISSUE THREAD

**Issue by jasondoc3** [klarna/phobos#33]
If a `delivery_interval` or `delivery_threshold` is desired on an async producer, the desired automatically delivery behavior is not achieved because `deliver_messages` is always called. See: https://github.com/klarna/phobos/blob/b28f6bdeb96ca3f84ffb35437e21efb257712541/lib/phobos/producer.rb

```rb
def produce_messages(producer, messages)
  messages.each do |message|
    producer.produce(message[:payload], topic: message[:topic],
                                        key: message[:key],
                                        partition_key: message[:key]
    )
  end
  producer.deliver_messages
end

```

**Comment by tulios** [comment]
You are right, we can check if the producer has any of the keys configured and conditionally call `deliver_messages`.

Thanks

**Comment by klippx** [comment]
Released in 1.4.2.
